src/ethereum-json-rpc-client/src/outcall.rs

```diff
@@ -0,0 +1,45 @@
+use std::future::Future;
+use std::pin::Pin;
+
+use ic_exports::ic_cdk::api::management_canister::http_request::{
+    CanisterHttpRequestArgument, HttpMethod, HttpResponse as MHttpResponse,
+};
+
+/// http outcall argument
+#[derive(Debug)]
+pub struct HttpOutcallArgs {
+    pub url: String,
+    pub method: HttpMethod,
+    pub body: Option<Vec<u8>>,
+    /// Max response from the http outcall
+    ///
+    /// # NOTE
+    /// As much as this is optional, it is important to set the value
+    /// otherwise it will be set to default 2MiB which uses a lot of cycles
+    pub max_response_bytes: Option<u64>,
```

Comment:
this should probably be not public and calculated automatically when the HttpCallArgs are created (e.g. by using a builder pattern) or just before the call is performed

Reply:
How can it be calculated automatically ? The http outcall doesn't know the response, it is upon the caller to calculate and estimate the response and use it here, otherwise it will use the default 2 MiB.

Reply:
indeed, you are right. I was confused.